crates/matrix-sdk-base/src/store/mod.rs

```diff
@@ -139,12 +140,63 @@ pub(crate) struct Store {
     /// The current sync token that should be used for the next sync call.
     pub(super) sync_token: Arc<RwLock<Option<String>>>,
     /// All rooms the store knows about.
-    rooms: Arc<StdRwLock<BTreeMap<OwnedRoomId, Room>>>,
+    rooms: Arc<StdRwLock<Rooms>>,
     /// A lock to synchronize access to the store, such that data by the sync is
     /// never overwritten.
     sync_lock: Arc<Mutex<()>>,
 }
 
+#[derive(Default, Debug)]
+struct Rooms {
```

Comment:
nit: really against this name; sounds like a variable name, not a type one can use; plus it would be really hard to find easily using `ripgrep`-like tools. Can you think of another name?

Reply:
It's been renamed `ObservableMap` later.